From 65019ef022a916948b4b560179c5a4a38ac32070 Mon Sep 17 00:00:00 2001
From: Alexei Colin <alexei.co@gmail.com>
Date: Sun, 16 Dec 2012 22:19:58 -0500
Subject: [PATCH] Connector 2.1: expose attachmentMode in saveItems

This allows one to add files by path via the connector.

Conflicts:

	chrome/content/zotero/xpcom/server_connector.js
---
 chrome/content/zotero/xpcom/server_connector.js |   19 +++++++++++++++++--
 1 file changed, 17 insertions(+), 2 deletions(-)

diff --git a/chrome/content/zotero/xpcom/server_connector.js b/chrome/content/zotero/xpcom/server_connector.js
index f52633b..f07909d 100644
--- a/chrome/content/zotero/xpcom/server_connector.js
+++ b/chrome/content/zotero/xpcom/server_connector.js
@@ -22,7 +22,7 @@
     
     ***** END LICENSE BLOCK *****
 */
-const CONNECTOR_API_VERSION = 2;
+const CONNECTOR_API_VERSION = 2.1;
 
 Zotero.Server.Connector = function() {};
 Zotero.Server.Connector._waitingForSelection = {};
@@ -312,6 +312,9 @@ Zotero.Server.Connector.SavePage.prototype = {
  *
  * Accepts:
  *		items - an array of JSON format items
+ *		attachmentMode - add items as files ('file') or download ('download'),
+ *					     optional, default is 'download'
+ *					     (see ItemSaver.ATTACHMENT_MODE_*)
  * Returns:
  *		201 response code with item in body.
  */
@@ -340,13 +343,25 @@ Zotero.Server.Connector.SaveItem.prototype = {
 		
 		var cookieSandbox = data["uri"] ? new Zotero.CookieSandbox(null, data["uri"],
 			data["cookie"] || "", url.userAgent) : null;
+
+		var attachmentMode =
+			Zotero.Translate.ItemSaver.ATTACHMENT_MODE_DOWNLOAD;
+		var requestedAttachmentMode = data["attachmentMode"];
+		if (requestedAttachmentMode && requestedAttachmentMode.toUpperCase) {
+			var attachmentModeName =
+				"ATTACHMENT_MODE_" + requestedAttachmentMode.toUpperCase();
+			if (Zotero.Translate.ItemSaver[attachmentModeName]) {
+				attachmentMode = Zotero.Translate.ItemSaver[attachmentModeName];
+			}
+		}
+
 		for(var i=0; i<data.items.length; i++) {
 			Zotero.Server.Connector.AttachmentProgressManager.add(data.items[i].attachments);
 		}
 		
 		// save items
 		var itemSaver = new Zotero.Translate.ItemSaver(libraryID,
-			Zotero.Translate.ItemSaver.ATTACHMENT_MODE_DOWNLOAD, 1, undefined, cookieSandbox);
+			attachmentMode, 1, undefined, cookieSandbox);
 		itemSaver.saveItems(data.items, function(returnValue, items) {
 			if(returnValue) {
 				try {
-- 
1.7.9.5

